platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -681,28 +717,54 @@ export default class HangingProtocolService extends PubSubService {
    * simpler than recomputing the entire protocol.
    */
   public getComputedOptions(
-    options: Record<string, unknown>,
+    options: Record<string, unknown> | Array<Record<string, unknown>>,
     displaySetUIDs: string[]
-  ) {
-    const computed = { ...options };
-    let displaySets;
-    for (const key in computed) {
-      const value = computed[key];
-      if (!value) continue;
-      if (value.custom) {
-        if (!displaySets) {
-          displaySets = this.displaySets.filter(
-            displaySet =>
-              displaySetUIDs.indexOf(displaySet.displaySetInstanceUID) !== -1
-          );
-        }
-        computed[key] = this.customAttributeRetrievalCallbacks[
-          value.custom
-        ].callback.call(computed, displaySets);
-        if (computed[key] === undefined) computed[key] = computed.defaultValue;
+  ): any {
+    // Base case: if options is an array, map over the array and recursively call getComputedOptions
+    if (Array.isArray(options)) {
+      return options.map(option =>
+        this.getComputedOptions(option, displaySetUIDs)
+      );
+    }
+
+    if (options === null) return options;
+    if (typeof options !== 'object') return options;
+
+    // If options is an object with a custom attribute, compute a new options object
+    if (options.custom) {
+      const displaySets = this.displaySets.filter(displaySet =>
+        displaySetUIDs.includes(displaySet.displaySetInstanceUID)
+      );
+
+      const customKey = options.custom as string;
+      if (!(customKey in this.customAttributeRetrievalCallbacks)) {
+        throw new Error(
+          `Custom key "${customKey}" not found in customAttributeRetrievalCallbacks.`
+        );
+      }
+
+      const callback = this.customAttributeRetrievalCallbacks[customKey]
+        .callback;
+      const newOptions = callback.call(options, displaySets);
+
+      if (newOptions === undefined) {
+        return options.defaultValue;
```

Comment:
How about:
newOptions = options.defaultValue
so that you have the option of having nested computed values in the response.